pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -433,6 +433,35 @@ type GarbageCollector struct {
 	propagator  *Propagator
 }
 
+// TODO: make special List and Watch function that removes fields other than
+// ObjectMeta.
+func gcListWatcher(client *dynamic.Client, resource unversioned.GroupVersionResource) *cache.ListWatch {
+	return &cache.ListWatch{
+		ListFunc: func(options api.ListOptions) (runtime.Object, error) {
+			// APIResource.Kind is not used by the dynamic client, so
+			// leave it empty. We want to list this resource in all
+			// namespaces if it's namespace scoped, so leave
+			// APIResource.Namespaced as false is all right.
+			apiResource := unversioned.APIResource{Name: resource.Resource}
+			// The default parameter codec used by the dynamic client cannot
+			// encode api.ListOptions.
+			// TODO: api.ParameterCodec doesn't support thirdparty objects.
+			// We need a generic parameter codec.
+			return client.Resource(&apiResource, api.NamespaceAll).List(&options, api.ParameterCodec)
```

Comment:
Hard coding what? The codec? Because we don't want that dependency in the dynamic client.